com.reprezen.swagedit/src/com/reprezen/swagedit/editor/PathParamHyperlinkDetector.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+ * Copyright (c) 2016 ModelSolv, Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ModelSolv, Inc. - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package com.reprezen.swagedit.editor;
+
+import static com.google.common.base.Strings.emptyToNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.ITextViewer;
+import org.eclipse.jface.text.Region;
+import org.eclipse.jface.text.hyperlink.IHyperlink;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import com.reprezen.swagedit.json.JsonUtil;
+
+import io.swagger.models.HttpMethod;
+
+/**
+ * Hyperlink detector that detects links from path parameters.
+ *
+ */
+public class PathParamHyperlinkDetector extends AbstractSwaggerHyperlinkDetector {
+
+	protected static final Pattern PARAMETER_PATTERN = Pattern.compile("\\{(\\w+)\\}");
+
+	@Override
+	public IHyperlink[] detectHyperlinks(ITextViewer textViewer, IRegion region, boolean canShowMultipleHyperlinks) {
+		SwaggerDocument document = (SwaggerDocument) textViewer.getDocument();
+
+		String basePath;
+		try {
+			basePath = document.getPath(region);
+		} catch (BadLocationException e) {
+			basePath = null;
+		}
+
+		// not a path definition
+		if (emptyToNull(basePath) == null || !basePath.startsWith(":paths:")) {
+			return null;
+		}
+
+		HyperlinkInfo info;
+		try {
+			info = getHyperlinkInfo(textViewer, region);
+		} catch (BadLocationException e) {
+			return null;
+		}
+
+		if (info == null) {
+			return null;
+		}
```

Comment:
These 11 lines related retrieval of `HyperLink` info are the same in the subclasses